hikari/traits.py

```diff
@@ -446,4 +446,129 @@ async def update_presence(
 class BotAware(RESTAware, ShardAware, EventFactoryAware, DispatcherAware, typing.Protocol):
     """Structural supertype for a component that is aware of all internals."""
 
-    __slots__: typing.Sequence[str] = ()
+    async def join(self, until_close: bool = True) -> None:
+        """Wait indefinitely until the application closes.
```

Comment:
```suggestion
class BotAware(RESTAware, ShardAware, EventFactoryAware, DispatcherAware, typing.Protocol):
    """Structural supertype for a component that is aware of all internals."""

    __slots__: typing.Sequence[str] = ()
    
    async def join(self, until_close: bool = True) -> None:
        """Wait indefinitely until the application closes.
```

Reply:
Gah, GitHub won't let me do it. Would you mind just re-adding the slots in for me here? You'd moved them down below everything else and I was trying to undo this for you so you didn't have to waste time fixing stuff that wasn't your fault, but I guess I made it worse. Sorry...